clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,239 @@
+    }
+
+    pub fn meshnet_log() -> String {
+        format!("{}/meshnet.log", Self::QPKG_DIR)
+    }
+
+    pub fn pid() -> String {
+        format!("{}/teliod.pid", Self::APP_TMP_DIR)
+    }
+}
+
+#[derive(Debug, PartialEq)]
+enum HttpMethod {
+    Get,
+    Post,
+    Patch,
+    Unsupported,
+}
+
+impl HttpMethod {
+    fn from_method(method: &str) -> Self {
+        match method {
+            "GET" => HttpMethod::Get,
+            "POST" => HttpMethod::Post,
+            "PATCH" => HttpMethod::Patch,
+            _ => HttpMethod::Unsupported,
+        }
+    }
+
+    fn get_post_data(&self) -> String {
+        let mut data = String::new();
+        if matches!(self, HttpMethod::Post | HttpMethod::Patch) {
+            let content_length = env::var("CONTENT_LENGTH")
+                .ok()
+                .and_then(|len| len.parse::<usize>().ok());
+            if let Some(length) = content_length {
+                let _ = io::stdin().take(length as u64).read_to_string(&mut data);
+            }
+        }
+        data
+    }
+}
+
+#[derive(Debug, PartialEq)]
+enum Action {
+    Start,
+    Stop,
+    UpdateConfig,
+    GetStatus,
+    GetTeliodLogs,
+    GetMeshnetLogs,
+    Invalid,
+}
+
+impl Action {
+    fn from_query(query: &str) -> Self {
+        match query {
+            "action=start" => Action::Start,
+            "action=stop" => Action::Stop,
+            "action=update-config" => Action::UpdateConfig,
+            "action=get-status" => Action::GetStatus,
+            "action=get-teliod-logs" => Action::GetTeliodLogs,
+            "action=get-meshnet-logs" => Action::GetMeshnetLogs,
+            _ => Action::Invalid,
+        }
+    }
+}
+
+pub(crate) fn parse_cgi() {
+    println!("Content-Type: application/json\n");
+
+    let http_method = {
+        let method = env::var("REQUEST_METHOD").unwrap_or_default();
+        HttpMethod::from_method(&method)
+    };
+    let post_data = http_method.get_post_data();
+    let action = {
+        let query = env::var("QUERY_STRING").unwrap_or_default();
+        Action::from_query(&query)
+    };
+
+    match (http_method, action) {
+        (HttpMethod::Post, Action::Start) => start_daemon(),
+        (HttpMethod::Post, Action::Stop) => stop_daemon(),
+        (HttpMethod::Patch, Action::UpdateConfig) => update_config(&post_data),
+        (HttpMethod::Get, Action::GetStatus) => get_status(),
+        (HttpMethod::Get, Action::GetMeshnetLogs) => get_meshnet_logs(),
+        (HttpMethod::Get, Action::GetTeliodLogs) => get_teliod_logs(),
+        (HttpMethod::Unsupported, _) => send_response(405, "Method not allowed."),
+        (_, Action::Invalid) => send_response(400, "Invalid action."),
+        _ => send_response(400, "Bad request."),
+    }
+}
+
+fn is_teliod_running() -> bool {
+    if let Ok(pid) = fs::read_to_string(PathConfig::pid()) {
+        if fs::metadata(format!("/proc/{}", pid.trim())).is_ok() {
+            return true;
+        }
+    }
+    false
```

Comment:
It could be just `.map` + `unwrap_or(false)`

Reply:
can you provide an example?

Reply:
Sth like this, I guess:
```fn is_teliod_running() -> bool {
    fs::read_to_string(PathConfig::pid())
        .and_then(|pid| fs::metadata(format!("/proc/{}", pid.trim())))
        .is_ok()
}
```

But tbh we agreed to use existance of the socket instead: https://github.com/NordSecurity/libtelio/blob/c60eedeeeca8e5f0928bc04ea29c9f720806cdff/clis/teliod/src/main.rs#L96 soo, maybe it should be made consistent?

Reply:
thanks for the example, I'll use it.

> But tbh we agreed to use existance of the socket instead: 

I don't see the benefits of using it, we still need to keep track of the PID to be able to stop the application. Maybe that suits the checks before socket communications but not here I think.